js_modules/dagster-ui/packages/ui-core/src/app/Flags.tsx

```diff
@@ -37,16 +37,15 @@ const initializeFeatureFlags = () => {
 
 /**
  * Internal function to set feature flags without broadcasting.
- * Used during initialization and migration.
+ * Used during initialization and migration and by web-workers.
  */
-const setFeatureFlagsInternal = (flags: FeatureFlagMap, broadcast: boolean = true) => {
+export const setFeatureFlagsInternal = (flags: FeatureFlagMap) => {
   if (typeof flags !== 'object' || Array.isArray(flags)) {
     throw new Error('flags must be an object mapping FeatureFlag to boolean values');
   }
   currentFeatureFlags = flags;
-  localStorage.setItem(DAGSTER_FLAGS_KEY, JSON.stringify(flags));
-  if (broadcast) {
-    featureFlagsChannel.postMessage('updated');
+  if (typeof localStorage !== 'undefined') {
```

Comment:
Might be nice to call out that this is for the webworker case just so we don't forget 😬